conn.go

```diff
@@ -731,6 +748,13 @@ type callReq struct {
 	streamID int           // current stream in use
 
 	timer *time.Timer
+
+	// streamObserverContext is notified about events regarding this stream
+	streamObserverContext StreamObserverContext
+
+	// streamObserverOnce ensures that either StreamAbandoned or StreamFinished is called,
+	// but not both.
+	streamObserverOnce sync.Once
```

Comment:
Nice solution with `sync.Once`! I only found the field name slightly misleading in the code. WDYT about renaming it to `streamEndObserverOnce` (or anything alse) to make it understandable from the first glance, that we are limiting calls only at the end of the stream lifecycle.

```suggestion
	streamEndObserverOnce sync.Once
```

Reply:
Renamed to `streamObserverEndOnce`, thanks!